Ansible/roles/marvin/templates/smoketests.sh.j2

```diff
@@ -35,32 +35,92 @@ rm -f /usr/lib/python2.7/site-packages/marvin/config/*pyc
 rm -f /usr/lib/python2.7/site-packages/marvin/lib/*pyc
 
 cleanup() {
+    # dont wait for async job - execute in parallel until told otherwise
     cloudmonkey set asyncblock false
-    cloudmonkey set display default
-    # TODO: destroy accounts, domains, templates
-    # Destroy vms
-    for vmid in $(cloudmonkey list virtualmachines listall=true filter=id | grep '^id' | awk '{print $3}'); do
-        cloudmonkey destroy virtualmachine id=$vmid expunge=true || true
+    cloudmonkey set output json
+    # TODO: N/A
+
+    # Destroy alerts
+    for id in $(cloudmonkey list alerts listall=true keyword=vm filter=id | jq -r '.alert[] .id'); do
+        cloudmonkey delete alerts ids=$id || true
+    done
+
+    # Destroy templates, forced, since some VMs might exist.
+    for templateid in $(cloudmonkey list templates templatefilter=self filter=id | jq -r '.template[] .id'); do
+        cloudmonkey delete template id=$templateid forced=true  || true
     done
-    # Destroy routers
-    for rid in $(cloudmonkey list routers listall=true filter=id | grep '^id' | awk '{print $3}'); do
-        cloudmonkey destroy router id=$rid || true
+
+    # Destroy vms, all in parallel
+    for vmid in $(cloudmonkey list virtualmachines listall=true filter=id | jq -r '.virtualmachine[] .id'); do
+        cloudmonkey destroy virtualmachine id=$vmid expunge=true || true
     done
-    # Destroy networks
-    for netid in $(cloudmonkey list networks listall=true filter=id | grep '^id' | awk '{print $3}'); do
+
+    # Not needed to destroy VRs explicitely, will be destroyed with network/VPC
+
+    # Destroy networks only when no more VMS (otherwise will fail if network contains a VM)
+         NoVMs=$(cloudmonkey list virtualmachines filter=id  | grep id | wc -l)
+         while [ $NoVMs -ne 0 ]
```

Comment:
is some mechanism required to break out of this loop if a vm expunge fails, otherwise this could get stuck in an infinite loop.

Reply:
Yes, it will be stuck, I can try to set some counter/time limit in the while loop (2 minutes)?
FTR - previous/current version of the cleanup "script" was not even able to delete resources since all API calls are executed at once (asyncblock=fales) - and we can't delete networks while it have VMs, can't delete VPCs while it has netowrks etc..

Reply:
Addressed.